transport/tchannel/transport.go

```diff
@@ -229,7 +237,7 @@ func (t *Transport) start() error {
 			excludeServiceHeaderInResponse: t.excludeServiceHeaderInResponse,
 		},
 		OnPeerStatusChanged: t.onPeerStatusChanged,
-		Dialer:              t.dialer,
+		Dialer:              dialer,
```

Comment:
if we don't pass `EnableMPTCP:true `here, and` t.dialer `is nil, ` chopts` will generate a channel without MPTCP dialer. Hence I modified this part.